9/README.md

```diff
@@ -303,6 +303,10 @@ DevPipeBlob is a Data Type to transfer data related to Pipes (see 7/Pipe).
  inner-blob-name = name
 ```
 
+Where \<value\> SHALL be a value of one of the following data types: DevVarBooleanArray, DevVarShortArray, DevVarLongArray,			
+DevVarLong64Array, DevVarFloatArray, DevVarDoubleArray, DevVarCharArray, DevVarUShortArray,	DevVarULongArray,		
+DevVarULong64Array, DevVarStringArray, DevVarStateArray, DevState, DevVarEncodedArray, DevBoolean.
```

Comment:
What about scalar DevDouble, DevFloat, ...?

Reply:
Apparently, according to IDL, there are no scala data types. It is [AttrValUnion](https://github.com/tango-controls/tango-idl/blob/1e5edb84d966814ad367f2674ac9a5658b6724ac/tango.idl#L363).

Reply:
According to the IDL. Yes, you're right.
I've seen the doc (https://tango-controls.readthedocs.io/en/latest/development/client-api/cpp-client-programmers-guide.html#reading-writing-device-pipe) is mentioning scalar data types too.
I think scalar are simply transported in arrays.

There is something confusing, I think.
It is the presence of DevBoolean type.
It is there only for the ATT_NO_DATA case.

Reply:
Yes, DevBoolean is only to poit there are no data :-)

Reply:
[DevicePipeBlob::get_data_elt_type() method in cppTango](https://github.com/tango-controls/cppTango/blob/tango-9-lts/cppapi/client/devapi_pipe.cpp#L508-L517) distinguishes scalar and spectrum types.
So, from the user point of view, it seems like the data elements can be scalar Tango data types too.

Reply:
And when looking deeper into cppTango code, it seems like the inner_blob_name is set to "scalar" string when the element is a scalar.

https://github.com/tango-controls/cppTango/blob/tango-9-lts/cppapi/client/devapi_pipe.cpp#L507-L513

Reply:
Yes, you are right. I think what matters here is users' perspectives not a corba implementation. I will add the missing types.

Reply:
It is interesting to note that the corresponding [JTango PipeDataElement::getType](https://github.com/tango-controls/JTango/blob/3c826dc478e1a4c2f3c6f2d154393184801f3e29/common/src/main/java/fr/esrf/TangoApi/PipeDataElement.java#L441) method does not seem to distinguish scalar and array types.

The PipeDataElement constructors seem to construct some DevPipeDataElt objects which are using a parameter which can be SCALAR or ARRAY, so this seems to be important at some point in Java too.

Reply:
I have added the [scalars](https://github.com/tango-controls/rfc/pull/61/commits/d804c9f03fa5a062695a9b359b763a0fc8884ce9)

Reply:
Should we create a list of potential pitfalls discovered during writing RFCs?

Reply:
Why not? Where do you want to store them?

Reply:
I don know yet